netifi-broker-client/src/main/java/com/netifi/broker/DefaultBrokerService.java

```diff
@@ -457,52 +457,83 @@ private synchronized void createConnection() {
     }
   }
 
+  private boolean attemptInitialConnection = true;
+
   public RSocket selectRSocket() {
     RSocket rSocket;
     List<WeightedReconnectingRSocket> _m;
     int r;
     for (; ; ) {
-      boolean createConnection;
+      final int size;
       synchronized (this) {
         r = missed;
         _m = members;
 
-        createConnection = members.size() < selectRefresh;
-      }
+        int _s = _m.size();
 
-      if (createConnection) {
-        createConnection();
-        continue;
+        boolean _a = false;
+        if (attemptInitialConnection && _s == 0) {
+          attemptInitialConnection = false;
+          _a = true;
+        }
+
+        size =
+            _a
+                    || (_s < poolSize
+                        && (System.currentTimeMillis() - selectRefreshTimeout)
```

Comment:
I don't see where selectRefreshTimeout gets updated. Is it meant to be a one-time 10s timeout?

Reply:
One time deal -  this is the part that should be replaced with a tombstone...

Reply:
Understood.